src/app/product.service.ts

```diff
@@ -10,4 +10,12 @@ export class ProductService {
   create(product) {
     return this.database.list('/products').push(product);
   }
+
+  getAll() {
+    return this.database.list('/products').snapshotChanges();
+  }
+
+  getProduct(productId) {
+    return this.database.object('/products/' + productId);
```

Comment:
at the end of this is where you should be putting the .valueChanges()

Reply:
Done.